### PR TITLE
Impro529 recursive filter bug

### PR DESCRIPTION
**Pull request by cgsandford**
Fixed x / y coordinate confusion in recursive filter.

Testing:
 - [ ] Ran tests and they passed OK
 - [ ] Added new tests for the new feature(s)

### CODE REVIEW

File: lib/improver/tests/nbhood/nbhood/test_RecursiveFilter.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
 import numpy as np
 from improver.nbhood.recursive_filter import RecursiveFilter
 from improver.nbhood.square_kernel import SquareNeighbourhood
+from improver.utilities.cube_manipulation import enforce_coordinate_ordering
```
[the remark this answers is not in the document]
Comment (cgsandford) [reply lib/improver/tests/nbhood/nbhood/test_RecursiveFilter.py]: Done.

File: tests/improver-nbhood/10-recursive-filter-alphas-basic-square.bats [hunk 2]
```diff
@@ -45,7 +45,7 @@
 
   # Run nccmp to compare the output and kgo.
   improver_compare_output "$TEST_DIR/output.nc" \
-      "$IMPROVER_ACC_TEST_DIR/nbhood/recursive/kgo_recursive_alphas.nc"
+      "$IMPROVER_ACC_TEST_DIR/nbhood/recursive/kgo_recursive_alphas_gridded.nc"
```
Comment (carolinejjones) [inline tests/improver-nbhood/10-recursive-filter-alphas-basic-square.bats]: This is a better name but we must remember to delete the old version.

File: lib/improver/nbhood/recursive_filter.py [hunk 3]
```diff
@@ -225,23 +162,28 @@ def recurse_backwards_y(grid, alphas):
 
         Args:
             grid (numpy array):
-                Array containing the input data to which the recursive filter
-                will be applied.
+                2D array containing the input data to which the recursive
+                filter will be applied.
```
[the remark this answers is not in the document]
Comment (cgsandford) [reply lib/improver/nbhood/recursive_filter.py]: Done.

**Comment by carolinejjones** [comment]
Rerun tests - all looks OK.

**Pull request merged**